ansible_runner/utils/capacity.py

```diff
@@ -22,16 +27,28 @@ def get_mem_in_bytes():
 
 
 def get_uuid():
-    try:
-        with open('/var/lib/dbus/machine-id') as f:
-            uuid = f.read()
-        return uuid.strip()
-    except FileNotFoundError:
-        try:
-            with open('/etc/machine-id') as f:
-                uuid = f.read()
-                return uuid.strip()
-        except FileNotFoundError:
-            error = ("Could not find /var/lib/dbus/machine-id or "
-                     "/etc/machine-id files, UUID undiscoverable.")
-            return error
+    uuid_file_path = Path('/etc/ansible/facts.d/uuid.txt')
+    if uuid_file_path.exists():
+        # Read the contents of the uuid.txt file if it already exists
+        with open(uuid_file_path) as f:
+            saved_uuid = f.read()
+        return saved_uuid
+    else:
+        # Generate a new UUID if no uuid.txt file is found
+        newly_generated_uuid = _generate_uuid()
+        return newly_generated_uuid
+
+
+def _generate_uuid():
+    generated_uuid = str(uuid.uuid4())
+
+    # Store the newly-generated UUID in a new dir/file
+    uuid_dir = Path('/etc/ansible/facts.d')
+    uuid_dir.mkdir(parents=True, exist_ok=True)
+    uuid_file = 'uuid.txt'
+    uuid_file_path = uuid_dir / uuid_file
+
+    with uuid_file_path.open('w', encoding='utf-8') as uuid_file:
```

Comment:
What should the correct permissions be? How sensitive is this file?

The way Core (partially) defends against this type of race condition/attack is by the randomness of the parent temporary directory name, not the module payload file name. I'm not sure the payload that contains a module to be executed (in the case of Core) has the same risk profile as a file that contains a UUID. But I don't fully understand what a malicious actor could gain by substituting their own UUID for ours.

In the case of this file, we want it to be in a deterministic location, so randomizing the path and/or filename aren't really options. I suppose just setting the permissions to 400 would be enough?

That the default path is inside the user's home directory, and this code does not create any directories in the `uuid_file_path` also mitigates quite a bit of risk. The real trouble comes with `os.makedirs()` or `mkdir -p`, which isn't happening here.

Reply:
In any case, it's probably a good idea to be explicit about the permissions on the file rather than relying on the `umask`. Should this be 600, 640, or mainly just not world readable?

Reply:
I set the file to `600` by default, but the caller can change it.